### DISCUSSION TITLE
Styles depending on value of an enum

### DISCUSSION THREAD

**Discussion by tisis2** [AvaloniaUI/Avalonia#12087] [Q&A] [answered]
Hi there,

maybe anyone of you got an idea how this small code sample could be simplified with the possibilities Avalonia offers.
a central point is that I try to really separate the view model and view code in separate projects and do not want to get any dependency on Avalonia in my view model project.
the project contains some menu entries and the view project should decide how a menu entry is presented and what icon it gets. so I created styles for the menu buttons that should decide which icon to show depending on the enum value of the MenuEntry view model. Since you cannot style an item depending on a property of the data context, I created bindings on classes and styled the button/image depending on the classes. somehow this feels to be not really the correct and smartest way to reach the goal, since you have to add new Classes binding and the style for the class.
please keep in mind that the example is an abstracted simplified version to showcase the issue. in the real project there is not only the image content oft the button that gets styled. there is also e.g. the menu text that is translated in view because in our opinion the translation should also not be done in a viewmodel.

i thought of something like a style selector like `Button[DataContext.Type=MenuEntryType.X] Image` but this is not possible.

[MenuEntries.zip](https://github.com/AvaloniaUI/Avalonia/files/11980119/MenuEntries.zip)

i look forward to your creative ideas and thanks for your help.

best regards
Dominik Gabriel

**Comment by timunie** [comment]
I feel like I'd not use style classes for this kind of stuff. I'd rather use one of those:

a) Use a ValueConverter that returns the image based on the enum or
b) write a MarkupExtension to do it or
c) implement a custom IDataTemplate to provide the needed DataTemplate or
d) use Avalonia Behaviors   

If you want to have more input on one of the ideas, let me know the letter and I'll try to provide a useful link

**Reply by tisis2** [reply, marked as answer]
thanks for your answer. i tried the converter thing and ended up with the following 
[MenuEntries_2.zip](https://github.com/AvaloniaUI/Avalonia/files/12024729/MenuEntries_2.zip)

**Reply by timunie** [reply]
LGTM 👍
